syntax = "proto3";

package siswa;

message Siswa {
  int32 id = 1;
  string nama = 2;
  float nilai = 3;
}

service SiswaService {
  rpc TambahSiswa(Siswa) returns (Siswa);
  rpc DapatkanSiswaById(int32) returns (Siswa);
  rpc PerbaruiSiswa(Siswa) returns (Siswa);
  rpc HapusSiswa(int32) returns (Siswa);
  rpc SemuaSiswa(Empty) returns (stream Siswa);
}

message Empty {}
